packages/lib/apps/getTeamAppCredentials.ts

```diff
@@ -0,0 +1,38 @@
+import type { Prisma } from "@prisma/client";
+
+import type { TeamQuery } from "@calcom/trpc/server/routers/loggedInViewer/integrations.handler";
+import type { CredentialPayload } from "@calcom/types/Credential";
+
+export type Credentials = {
+  user: {
+    email: string;
+  } | null;
+  id: number;
+  userId: number | null;
+  type: string;
+  key: Prisma.JsonValue;
+  teamId: number | null;
+  appId: string | null;
+  invalid: boolean | null;
+}[];
+
+const getTeamAppCredentials = (
+  userTeams: TeamQuery[],
+  userCredentialIds: Credentials,
```

Comment:
What do you think about this? Maybe merge outside of this function to keep this function dumber?

Reply:
Agreed, maybe we can create a separate function for that?